fearless/Common/Repository/UtilsLocalRepository.swift

```diff
@@ -0,0 +1,83 @@
+import Foundation
+import RobinHood
+
+final class UtilsLocalRepository<T: Codable & Equatable & Identifiable> {
+    private let url: URL
+    private let logger: LoggerProtocol?
+    private let repository: AnyDataProviderRepository<SingleValueProviderObject>
+    private var cache: T?
+
+    private var filename: String? {
+        guard let filename = url.absoluteString.sha256String() else {
+            return nil
+        }
+
+        return URL.documentsDirectoryUrl().appendingPathComponent(filename).path
+    }
+
+    init(url: URL, logger: LoggerProtocol?, repository: AnyDataProviderRepository<SingleValueProviderObject>) {
+        self.url = url
+        self.logger = logger
+        self.repository = repository
+
+        cache = readFromRepository()
+
+        load { [weak self] result in
+            switch result {
+            case let .success(resultData):
+                self?.cache = resultData
+            case let .failure(error):
+                logger?.error(error.localizedDescription)
+            }
+        }
+    }
+
+    func fetch() -> T? {
+        cache
+    }
+
+    private func readFromRepository() -> T? {
```

Comment:
so basically, if url was once loaded it's stuck in cache forever? with no overwrite even if changes are done?

Reply:
@posplaw no. every time you request url, it updates from remote repository and overwrite cached data.